File: Week1/GuessMyNumber.js
"use strict";

window.addEventListener("DOMContentLoaded", main);

function main() {
  const log = document.querySelector("#log");
  const secretInput = document.querySelector("#secret");
  const lockBtn = document.querySelector("#lockBtn");
  const guessBtn = document.querySelector("#guessBtn");

  let secret = null;
  let tries = 0;

  lockBtn.addEventListener("click", () => {
    const val = Number(secretInput.value);
    if (!Number.isInteger(val) || val < 1 || val > 100) {
      addLine("Vælg et tal mellem 1 og 100 først.", "error");
      return;
    }
    secret = val;
    tries = 0;
    clearLog();
    addLine("Dit tal er låst. Computeren begynder at gætte.");
    secretInput.disabled = true;
    lockBtn.disabled = true;
    guessBtn.disabled = false;
  });

  guessBtn.addEventListener("click", () => {
    if (secret === null) return;

    const guess = Math.floor(Math.random() * 100) + 1;
    tries++;

    if (guess < secret) {
      addLine(`Computeren gættede ${guess} – for lavt.`);
    } else if (guess > secret) {
      addLine(`Computeren gættede ${guess} – for højt.`);
    } else {
      addLine(`Computeren gættede ${guess} – korrekt!`);
      addLine(`Antal forsøg: ${tries}`);
      guessBtn.disabled = true;
    }
  });

  function addLine(text) {
    const li = document.createElement("li");
    li.textContent = text;
    log.appendChild(li);
  }

  function clearLog() {
    log.innerHTML = "";
  }
}

File: Week1/GuessMyNumber.java
package Week1;

import java.util.Random;
import java.util.Scanner;

public class GuessMyNumber {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        System.out.println("^^^^^^^^");
        System.out.println("Velkommen til 'Guess MY Number' ");
        System.out.print("Tænk på et tal mellem 1 og 100, og skriv det her: ");

        int secret = scanner.nextInt();
        scanner.nextLine(); // rydder linjen

        if (secret < 1 || secret > 100) {
            System.out.println("Tallet skal være mellem 1 og 100!");
            return;
        }

        System.out.println("\nOkay, jeg prøver at gætte dit tal...");

        int guess;
        int tries = 0;

        do {
            guess = random.nextInt(100) + 1; // 1-100
            tries++;
            System.out.println("Mit gæt #" + tries + ": " + guess);

            if (guess < secret) {
                System.out.println("For lavt!");
            } else if (guess > secret) {
                System.out.println("For højt!");
            } else {
                System.out.println(" Jeg gættede det! Dit tal var " + secret + "!");
                System.out.println("Jeg brugte " + tries + " forsøg.");
            }

        } while (guess != secret);

        System.out.println("Tak for spillet 👋");
        scanner.close();
    }
}